### PR TITLE
User Sign Up

### PR DESCRIPTION
**Pull request by erikwesterberg**
As a user,
In order to buy/sell/discuss plants,
I would like to be able to sign up to Florum
-Write unit tests for user registration (database)
-Write feature tests for user registration (sign-up @ page)
- Adds name to sign up form

Pivotaltracker - https://www.pivotaltracker.com/n/projects/2371102

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+    protect_from_forgery with: :exception
+
+    before_action :configure_permitted_parameters, if: :devise_controller?
+
+    protected
+
+         def configure_permitted_parameters
+              devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :email, :password)}
+
+              devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:name, :email, :password, :current_password)}
```
Comment (oliverochman) [inline app/controllers/application_controller.rb]: Out of scope, you are only working on `sign_up`. Not `account_update`

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -232,7 +232,7 @@
   # Turn scoped views on. Before rendering "sessions/new", it will first check for
   # "users/sessions/new". It's turned off by default because it's slower if you
   # are using only default views.
-  # config.scoped_views = false
+  config.scoped_views = true
```
Comment (oliverochman) [inline config/initializers/devise.rb:235]: can you explain what is happening here?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: config/initializers/devise.rb (hunk 2 above)
Comment (erikwesterberg) [reply config/initializers/devise.rb:235]: We had to change this to visually see the new name field on login.
Link - https://gist.github.com/withoutwax/46a05861aa4750384df971b641170407

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (erikwesterberg) [reply app/controllers/application_controller.rb]: Fixed mr mano

**Review by oliverochman — APPROVED** [review]

**Pull request merged**